docs/root/configuration/http_conn_man/headers.rst

```diff
@@ -299,3 +299,12 @@ Supported dynamic values are:
 %PROTOCOL%
     The original protocol which is already added by Envoy as a
     :ref:`x-forwarded-proto <config_http_conn_man_headers_x-forwarded-proto>` request header.
+
+%UPSTREAM_METADATA(namespace, key[, key...])%
+    Populates the header with ref:`EDS endpoint metadata <envoy_api_file_api/eds.proto>` from the
```

Comment:
We also have CDS metadata now that is an upstream concept. Do we want to include access to it in this header variable? Would we merge the metadatas together? We had this idea that the LDS/RDS metadata would all be merged before being made available to the user (at the RequestInfo level internally). I don't have a strong opinion on what it should look like, just raising these considerations.

Reply:
Ultimately, I think CDS metadata should get merged in, probably with EDS values taking precedence over CDS. It doesn't seem to have been plumbed  through to ClusterInfo yet, though.

Reply:
IMO I would just log a follow up issue for CDS metadata, but I could go either way.

Reply:
Filed envoyproxy/envoy#2184.